app/src/pages/redirects/spanRedirectLoader.ts

```diff
@@ -0,0 +1,41 @@
+import { fetchQuery, graphql } from "react-relay";
+import { LoaderFunctionArgs, redirect } from "react-router";
+
+import RelayEnvironment from "@phoenix/RelayEnvironment";
+
+import { spanRedirectLoaderQuery } from "./__generated__/spanRedirectLoaderQuery.graphql";
+
+export async function spanRedirectLoader({ params }: LoaderFunctionArgs) {
+  const { span_otel_id } = params;
+
+  if (!span_otel_id) {
+    throw new Error("Span redirect requires a span ID");
+  }
+
+  const response = await fetchQuery<spanRedirectLoaderQuery>(
+    RelayEnvironment,
+    graphql`
+      query spanRedirectLoaderQuery($spanOtelId: String!) {
+        span: searchSpanByOtelId(spanId: $spanOtelId) {
+          trace {
+            id
+            traceId
+          }
+          project {
+            id
+          }
+        }
+      }
+    `,
+    {
+      spanOtelId: span_otel_id,
+    }
+  ).toPromise();
+
+  if (response?.span) {
+    const { trace, project } = response.span;
+    return redirect(`/projects/${project.id}/spans/${trace.traceId}`);
```

Comment:
Is there a good heuristic for when we should instead redirect to traces page (if at all?) also can you add `?selectedSpanNodeId=${spanId}` to the url so that the actual span will be highlighted in the trace tree?